sklearn/compose/_column_transformer.py

```diff
@@ -136,6 +136,11 @@ class ColumnTransformer(TransformerMixin, _BaseComposition):
         sparse matrix or a dense numpy array, which depends on the output
         of the individual transformers and the `sparse_threshold` keyword.
 
+    transformers_output_ : dict
```

Comment:
not sure about this name.
Maybe just `transformer_slices_`? I'd prefer `transformer_outputs_` to `transformers_output_` as well.

Reply:
since we index the dict by transformer name already, I don't think we need the name to include `transformer`.  We do need to  indicate that this attribute gives indices (and not the actual output, as currently suggested). So personally I would prefer `output_indices_`

Reply:
I renamed the attribute to `transformer_slices_` for now. I am a bit reluctant about `indices`, since the term is often used to refer to a list of indices (i.e. `[0, 1, 2]` rather than `slice(0, 3)`), so it could be counterintuitive to get a slice instead. `transformer_slices_` has clarity going for it.

Reply:
I'm not a fan of `transformer_slices` because there's no notion of output or indexing in that name, which really is what this parameter is. I also suggests that the transformers are sliced, which isn't what's happening.

> I am a bit reluctant about indices, since the term is often used to refer to a list of indices (i.e. [0, 1, 2] rather than slice(0, 3)), so it could be counterintuitive to get a slice instead

As long as it's properly documented, it should be fine. Since slices can be used interchangeably with lists of indices in numpy arrays, there's little chance of misuse anyway. And it's always good to educate users ;)